src/key.rs

```diff
@@ -450,6 +451,12 @@ impl PublicKey {
             PublicKey(pk)
         }
     }
+    /// Creates a new public key from a [`ElligatorSwift`] and the global [`SECP256K1`] context.
+    #[inline]
+    pub fn from_ellswift(ellswift: ElligatorSwift) -> PublicKey {
+        let pk = ElligatorSwift::ellswift_decode(ellswift);
+        PublicKey(pk)
```

Comment:
```suggestion
        ElligatorSwift::ellswift_decode(ellswift)
```

And in `ellswift_decode` return a `PublicKey` using `PublicKey::from(pk)`.

Reply:
This only works if I change `PublicKey` because the inner `ffi::PublicKey` is private. Is it ok to do this?

```diff
- pub struct PublicKey(ffi::PublicKey);
+ pub struct PublicKey(pub(crate) ffi::PublicKey);
```

Reply:
Oh, nevermind. Just using `from` works.